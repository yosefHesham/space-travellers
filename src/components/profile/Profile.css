table tr:nth-child(even) {
  background-color: #e6e6e6;
}