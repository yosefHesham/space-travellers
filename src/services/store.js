import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import rocketsReducer from './rockets';

const rootReducer = combineReducers({
  rockets: rocketsReducer,
});

const store = createStore(rootReducer, applyMiddleware(thunk, logger));
export default store;
