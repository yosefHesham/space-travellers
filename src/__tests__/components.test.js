import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import JoinedMissions from '../components/profile/joinedMissions';
import Profile from '../components/profile/profile';
import ReservedRockets from '../components/profile/reservedRockets';
import store from '../redux/configure_store';
import Rockets from '../components/rockets/rockets';
import Missions from '../components/missions/missions';

const createComponent = (component) => <Provider store={store}>{component}</Provider>;

describe('Test for components', () => {
  test('Profile page should render correctly', () => {
    const component = renderer
      .create(createComponent(<Profile />))
      .toJSON();

    expect(component).toMatchSnapshot();
  });
  test('ReservedRockets should render correctly', () => {
    const component = renderer
      .create(createComponent(<ReservedRockets />))
      .toJSON();

    expect(component).toMatchSnapshot();
  });
  test('joinedMissions  should render correctly', () => {
    const component = renderer
      .create(createComponent(<JoinedMissions />))
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('rocket page should render properly', () => {
    const component = renderer.create(createComponent(<Rockets />)).toJSON();

    expect(component).toMatchSnapshot();
  });
  test('Missions page should render properly', () => {
    const component = renderer.create(createComponent(<Missions />)).toJSON();

    expect(component).toMatchSnapshot();
  });
});
