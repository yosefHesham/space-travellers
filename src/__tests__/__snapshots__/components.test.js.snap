// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for components Missions page should render properly 1`] = `
<section
  className="missions"
>
  <div
    className="mission"
  >
    <p
      className="right_border"
    >
      Mission
    </p>
    <p
      className="right_border"
    >
      Description
    </p>
    <p
      className="right_border"
    >
      Status
    </p>
    <p>
       
    </p>
  </div>
</section>
`;

exports[`Test for components Profile page should render correctly 1`] = `
<div
  className="profile_container"
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "margin": "30px auto",
      }
    }
  >
    <span
      style={
        Object {
          "margin": "20px 0",
          "padding": "0",
        }
      }
    >
      <h1>
        My Missions
      </h1>
    </span>
    <ul>
      <p>
        You have not reserved any missions 
      </p>
    </ul>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "margin": "30px auto",
      }
    }
  >
    <span
      style={
        Object {
          "margin": "20px 0",
          "padding": "0",
        }
      }
    >
      <h1>
        My Rockets
      </h1>
    </span>
    <ul>
      <p>
        You have not reserved any rockets 
      </p>
    </ul>
  </div>
</div>
`;

exports[`Test for components ReservedRockets should render correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "margin": "30px auto",
    }
  }
>
  <span
    style={
      Object {
        "margin": "20px 0",
        "padding": "0",
      }
    }
  >
    <h1>
      My Rockets
    </h1>
  </span>
  <ul>
    <p>
      You have not reserved any rockets 
    </p>
  </ul>
</div>
`;

exports[`Test for components joinedMissions  should render correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "margin": "30px auto",
    }
  }
>
  <span
    style={
      Object {
        "margin": "20px 0",
        "padding": "0",
      }
    }
  >
    <h1>
      My Missions
    </h1>
  </span>
  <ul>
    <p>
      You have not reserved any missions 
    </p>
  </ul>
</div>
`;

exports[`Test for components rocket page should render properly 1`] = `
<section
  style={
    Object {
      "padding": "0",
      "width": "100%",
    }
  }
>
  <ul
    style={
      Object {
        "padding": "0",
        "width": "100%",
      }
    }
  />
</section>
`;
